packages/amplication-client/src/Resource/git/ServiceConfigurationGitSettings.tsx

```diff
@@ -0,0 +1,141 @@
+import "./SyncWithGithubPage.scss";
+import "./ServiceConfigurationGitSettings.scss";
+import { ResourceWithGitRepository } from "./SyncWithGithubPage";
+import AuthResourceWithGit from "./AuthResourceWithGit";
+import ProjectConfigurationGitSettings from "./ProjectConfigurationGitSettings";
+import { AppContext } from "../../context/appContext";
+import { gql, useMutation } from "@apollo/client";
+import { UPDATE_RESOURCE } from "../ResourceForm";
+import * as models from "../../models";
+import { useTracking } from "../../util/analytics";
+import { DISCONNECT_GIT_REPOSITORY } from "./GitActions/RepositoryActions/GithubSyncDetails";
+
+const CLASS_NAME = "service-configuration-github-settings";
+
+type Props = {
+  resource: ResourceWithGitRepository;
+  onDone: () => void;
+};
+
+type TData = {
+  updateResource: models.Resource;
+};
+
+const ServiceConfigurationGitSettings: React.FC<Props> = ({
+  resource,
+  onDone,
+}) => {
+  const { currentWorkspace } = useContext(AppContext);
+  const [isOverride, setIsOverride] = useState<boolean>(
+    resource.gitRepositoryOverride
+  );
+
+  const { trackEvent } = useTracking();
+
+  const settingsClassName = isOverride
+    ? "gitSettingsPanel"
+    : "gitSettingsFromProject";
+
+    const [connectResourceToProjectRepository] = useMutation<TData>(CONNECT_RESOURCE_PROJECT_REPO, {
+      variables: { resourceId: resource.id }
+    });
+
+    const [disconnectGitRepository] = useMutation(DISCONNECT_GIT_REPOSITORY, {
+      variables: { resourceId: resource.id }
+    });
+  
+    const handleDisconnectGitRepository = useCallback(() => {
+      disconnectGitRepository({
+        variables: { resourceId: resource.id },
+      }).catch(console.error);
+    }, [disconnectGitRepository, resource.id]);
+
+
+    const handleConnectProjectGitRepository = useCallback(() => {
+      connectResourceToProjectRepository({
+        variables: { resourceId: resource.id },
+      }).catch(console.error);
+    }, [connectResourceToProjectRepository, resource.id]);
+
+    const handleResourceStatusChanged = useCallback(
+      (isOverride: boolean) => {
+      setIsOverride(isOverride);
+      if(isOverride) {
+        handleDisconnectGitRepository()
+      }
+      else {
+        handleConnectProjectGitRepository();
+      }
+    },
+    [handleDisconnectGitRepository,handleConnectProjectGitRepository]); 
+
+  const [updateResourceOverrideStatus] = useMutation<TData>(UPDATE_RESOURCE, {
+    onCompleted: (data) => {
+      handleResourceStatusChanged(data.updateResource.gitRepositoryOverride); 
+    },
+  });
+
+  const handleToggleChange = useCallback(
+    (gitRepositoryOverride) => {
+      trackEvent({
+        eventName: "updateResourceInfo",
+      });
+      updateResourceOverrideStatus({
+        variables: {
+          data: {
+            gitRepositoryOverride,
+          },
+          resourceId: resource.id,
+        },
+      }).catch(console.error);
```

Comment:
maybe you can refetch here, in `then` phase

Reply:
@barshimi  same here about the refetch